custom_reports/static/src/js/sales_by_company_graph.js

```diff
@@ -0,0 +1,221 @@
+        //                 label: 'ClickIt September Sale Ad',
+        //                 data: [27, 25, 29, 32, 30, 37, 55, 60, 65, 50, 30],
+        //                 borderColor: "#c45850",
+        //                 fill: false
+        //             },  {
+        //                 label: 'ClickIt October Sale Ad',
+        //                 data: [35, 29, 29, 32, 48, 50, 55, 60, 67, 70, 65],
+        //                 borderColor: "#3e95cd",
+        //                 fill: false
+        //             }]
+        //         },
+        //     };
+        // },
+
+        /**
+         * Returns a pie graph
+         * @private
+         */
+        _getPieGraph: function (labels, data, colors)   {
+            return {
+                type: 'pie',
+                data: {
+                   labels: labels,
+                   datasets: [{
+                       label: 'Sales by Company',
+                       data: data,
+                       backgroundColor: colors,
+                    }]
+                },
+            };
+        },
+
+        /**
+         * Returns a doughnut graph
+         * @private
+         */
+        _getDoughnutGraph: function (labels, data, colors)   {
+            return {
+                type: 'doughnut',
+                data: {
+                   labels: labels,
+                   datasets: [{
+                       label: 'Sales by Company',
+                       data: data,
+                       backgroundColor: colors,
+                    }]
+                },
+            };
+        },
+
+        /**
+         * Returns a polar graph
+         * @private
+         */
+        _getPolarGraph: function (labels, data, colors)   {
+            return {
+                type: 'polarArea',
+                data: {
+                   labels: labels,
+                   datasets: [{
+                       label: 'Sales by Company',
+                       data: data,
+                       backgroundColor: colors,
+                    }]
+                },
+            };
+        },
+
+        /**
+         * Returns a radar graph
+         * @private
+         */
+        _getRadarGraph: function (labels, data, colors)   {
+            return {
+                type: 'radar',
+                data: {
+                   labels: labels,
+                   datasets: [{
+                       label: 'Sales by Company',
+                       data: data,
+                       backgroundColor: colors,
+                    }]
+                },
+            };
+        },
+
+        /**
+         * Loads chart 
+         * @private
+         */
+        _loadChart: function()  {
+            var canvas = this.$('canvas')[0];
+            var context = canvas.getContext('2d');
+            return new Chart(context, this.config);
+        },
+
+        /**
+         * Returns an array of colors
+         * @private
+         */
+        _getColors: function(length)    {
```

Comment:
This has a similar problem as #50. There will limit the number of elements you can graph to 7.